app/src/lib/editors/launch.ts

```diff
@@ -13,6 +15,31 @@ export async function launchExternalEditor(
   editor: FoundEditor
 ): Promise<void> {
   const editorPath = editor.path
+
+  //Special launching for Xcode.
+  if (editor.editor === Darwin.Xcode) {
+    const projectFiles = await findFilesMatching('*.xcodeproj', fullPath)
+    const workspaces = await findFilesMatching('*.xcworkspace', fullPath)
```

Comment:
Since this is a relatively simple glob expression and there are two queries, we could potentially avoid hitting the filesystem twice and also avoid having to pull in an extra "glob" dependency:

```js
const files = await FSE.readdir(fullPath)
const projectFiles = files.filter(f => f.endsWith('.xcodeproj'))
const workspaces = files.filter(f => f.endsWith('.xcworkspace'))
```

Reply:
Hi, @mislav ! These sound like some great suggestions! I'll work on these now and commit to this PR today.